### PR TITLE
ocamlPackages.zmq: init at 20180726

### PR DESCRIPTION
**Pull request by akavel**
###### Motivation for this change

I plan to upstream an OCaml kernel for Jupyter Notebook later, and it has zmq as dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (Ubuntu 16.04 + Nix)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by akavel** [comment]
Newbie question: this mostly fits "CONTRIBUTING.md", except only having `meta.maintainers` set; whom should I put there? Should I add myself to the global list? Or add someone else here?

Also, I tested that it builds at `19.03pre153723.c1a66a49fb2`; should I test at some other revision? If yes, then how?

File: pkgs/development/ocaml-modules/zmq/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    repo = "ocaml-zmq";
+    rev = "d312a8458d6b688f75470248f11875fbbfa5bb1a";
+    sha256 = "1f5l4bw78y4drabhyvmpj3z8k30bill33ca7bzhr02m55yf6gqpf";
+  };
+
+  patches = [
+    ./ocaml-zmq-issue43.patch
+  ];
+
+  buildInputs = [ ocaml findlib dune stdint czmq ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/zmq/default.nix]: `stdint` should be propagated.

File: pkgs/development/ocaml-modules/zmq/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  buildPhase = "dune build -p zmq";
+
+  inherit (dune) installPhase;
+
+  meta = {
+    description = "ZeroMQ bindings for OCaml";
+    license = stdenv.lib.licenses.mit;
+    inherit (src.meta) homepage;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/zmq/default.nix]: please add a `platforms` attribute

File: pkgs/development/ocaml-modules/zmq/lwt.nix [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{ stdenv, ocaml, findlib, dune, zmq, lwt3, stdint }:
+
+stdenv.mkDerivation rec {
+  name = "ocaml${ocaml.version}-zmq-lwt-${version}";
+  inherit (zmq) version src installPhase meta;
+
+  buildInputs = [ ocaml findlib dune zmq lwt3 stdint ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/zmq/lwt.nix]: `zmq` and `lwt` should be propagated.

File: pkgs/development/ocaml-modules/zmq/lwt.nix [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{ stdenv, ocaml, findlib, dune, zmq, lwt3, stdint }:
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/zmq/lwt.nix]: Why did you choose lwt-3? Does it work with version 4?

**Review by vbgl — CHANGES_REQUESTED** [review]
Regarding maintainership, feel free to add yourself to the big list of maintainers!

File: pkgs/development/ocaml-modules/zmq/default.nix (hunk 1 above)
Comment (akavel) [reply pkgs/development/ocaml-modules/zmq/default.nix]: Ok, thanks! By the way, asking as a newbie: how can I discover/find out which packages should be put on the propagated list?

File: pkgs/development/ocaml-modules/zmq/default.nix (hunk 2 above)
Comment (akavel) [reply pkgs/development/ocaml-modules/zmq/default.nix]: Is it OK to just do the inherit here for that?

**Review by vbgl — APPROVED** [review]

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.zmq-lwt

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.zmq-lwt

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.zmq-lwt

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.zmq-lwt

**Pull request merged**